doc/barefoot_dtel/Dtel-SONiC.md

```diff
@@ -0,0 +1,509 @@
+and the switch is re-configured from scratch. euclid connects
+to the database container on the switch,
+and manipulates ConfigDB by adding, modifying, and deleting keys and values.
+
+All parameters are implemented as Python properties,
+so their values are set with simple assignment.
+In a typical workflow, a user would start by instantiating a switch:
+
+```python
+my_switch = sonic_switch.SONiCSwitch(dtel_switch_id=’123’,
+                                     management_ip=’10.10.10.10’,
+                                     dtel_monitoring_type='postcard')
+```
+
+Then create a report session:
+
+```python
+rs = my_switch.create_dtel_report_session(‘192.168.0.1’)
+```
+
+Create a watchlist:
+
+```python
+wl = my_switch.create_dtel_watchlist('flow')
+```
+
+Add entries to the watchlist:
+
+```python
+wl.create_entry(priority=10,
+                src_ip='10.131.0.0',
+                src_ip_mask=11,
+                dst_ip='10.131.0.0',
+                dst_ip_mask=11,
+                dtel_sample_percent=100,
+                dtel_report_all=True)
+```
+
+etc.
+
+To use euclid, users have to just import the module in their python script.
+Our tests for DTel make use of euclid, and more examples can be found there.
+
+Sample configuration in config_db.json after "config save":
+
+```
+    "DTEL_A_TABLE": {
+        "FLOW_STATE_CLEAR_CYCLE": {
+            "FLOW_STATE_CLEAR_CYCLE": "0"
+        },
+        "SINK_PORT_LIST": {
+            "Ethernet2": "Ethernet2",
+            "Ethernet0": "Ethernet0",
+            "Ethernet1": "Ethernet1"
+        },
+        "INT_ENDPOINT": {
+            "INT_ENDPOINT": "TRUE"
+        },
+        "INT_L4_DSCP": {
+            "INT_L4_DSCP_MASK": "1",
+            "INT_L4_DSCP_VALUE": "1"
+        },
+        "SWITCH_ID": {
+            "SWITCH_ID": "1"
+        },
+        "LATENCY_SENSITIVITY": {
+            "LATENCY_SENSITIVITY": "30"
+        }
+    },
+    
+    "DTEL_C_INT_SESSION_TABLE": {
+        "INT_SESSION1": {
+            "COLLECT_EGRESS_TIMESTAMP": "FALSE",
+            "COLLECT_SWITCH_PORTS": "FALSE",
+            "COLLECT_INGRESS_TIMESTAMP": "FALSE",
+            "COLLECT_SWITCH_ID": "TRUE",
+            "MAX_HOP_COUNT": "8",
+            "COLLECT_QUEUE_INFO": "FALSE"
+        }
+    },
+    
+    "DTEL_B_REPORT_SESSION_TABLE": {
+        "REPORT_SESSION1": {
+            "SRC_IP": "10.12.11.20",
+            "TRUNCATE_SIZE": "256",
+            "VRF": "default",
+            "DST_IP_LIST": "192.168.2.2",
+            "UDP_DEST_PORT": "32766"
+        }
+    },
+    
+    "ACL_RULE": {
+        "DTEL_FLOW_WATCHLIST|RULE1": {
+            "FLOW_SAMPLE_PERCENT": "100",
+            "PRIORITY": "10",
+            "ETHER_TYPE": "2048",
+            "REPORT_ALL_PACKETS": "TRUE",
+            "SRC_IP": "192.168.4.2/32",
+            "INT_SESSION": "INT_SESSION1",
+            "FLOW_OP": "INT",
+            "DST_IP": "192.168.3.2/32"
+        }
+    },
+```
+
+
+## Testing
```

Comment:
VirtualSwitch (a.k.a) VS testing should covers all swss parts including ACL changes as well as the new Dtel.
On top of that we have for each feature a testbed test which should cover system level tests and full functional validation with a real hw. this one should have dedicated test plan document and should be reviewed by community.
you have few examples of those in the github: crm, acl, etc.

Reply:
I will create a separate test plan document and upload it as part of this PR.